### ISSUE TITLE
Consider moving the BinderHub helm chart + deployment docs into the JupyterHub helm chart

### ISSUE THREAD

**Issue by choldgraf** [jupyterhub/team-compass#452]
For several reasons it would be useful long-term if there weren't two different helm charts (one for JupyterHub and one for BinderHub) as well as two totally separate sets of documentation, given that they overlap so much.

One idea that's been kicked around a bit is to treat a "BinderHub" as a particular flavor of a JupyterHub on Kubernetes, as opposed to its whole separate thing. Doing this would basically require two main things:

1. Make "BinderHub functionality" possible via configuration in the JupyterHub helm chart
2. Remove much of the "BinderHub install guide" docs into a section of the Z2JH guide

I think this could have a lot of benefits / improve the long-term sustainability of the codebases/documentation. It'd be great to chat about this further! Just opening this issue for conversation.

cc @yuvipanda as I think he has thoughts on this

**Comment by betatim** [comment]
Another idea: can we move to a model where you setup a JupyterHub and then deploy (one or more) extra services that use that JH? For BinderHub this would mean that the instructions would be "Use z2jh to deploy a JH, then deploy the BinderHub service (via its own chart)".

What are other JH services and how do they get deployed/added to a JH? Do we want to have all possible JH services controlled from the Z2JH chart? Can we model different services like plugins (not all controlled from a central repo)?

Right now the z2jh chart is a dependency of the BH chart. So if you already have a JupyterHub you can't add the BH service to it. Is this a problem that would be fixed/to which people desire a fix?

The writing I've read (from people outside the JH team) that dreams of a "Hub unification" seems to be based on the misunderstanding of what BH is. This means it is tricky to figure out what it is that people are desiring when they wish for a hub unification.

I understand what people desire from a "I am a *Hub developer" point of view, but I am wondering what features/benefits the  "*Hub users" are looking for. Getting a better understanding of those desires would be useful in order to decide what to do next. Without that we risk making a (big) move that still results in people being unhappy.

One thing I have found myself wanting, as a *Hub user, is being able to deploy one JH that is used like a normal JH by some users and then reuse that JH to also power a BH (that is deployed on the same cluster). Mostly because it is more work to deploy and maintain one JH for the "JH as normal" use case and then a second one for the "JH for BH" use case. However an upside has been that the configuration and operation of these two hubs is totally separate (I can upgrade them separately, changing settings for one doesn't touch the other).

TL;DR: I like the idea of thinking of adding services to a JH as "installing a plugin/extension" and I don't have a good idea what drives the desire of hub users for "hub unification".

**Comment by yuvipanda** [comment]
fwiw, I love @willingc's proposal in https://github.com/jupyterhub/team-compass/issues/452#issuecomment-371267846. At that time I had burnt out and toned down my involvement for a while, but I still strongly believe this is the path forward for better *adoption of binderhub*, which ultimately is something necessary for less stressful maintenance too.
